pypsa/optimization/constraints.py

```diff
@@ -437,11 +634,20 @@ def p_previous(idx: pd.Index) -> DataArray:
 
     # ----------------------------- Extendable Generators ----------------------------- #
 
+    ext_i = n.get_extendable_i(c)
+    com_i = n.get_committable_i(c)
+    mod_i = n.df(c).query(f"({nominal_attrs[c]}_mod>0)").index
+    inter_i = ext_i.difference(com_i).rename(ext_i.name)
+    inter_i = inter_i.union(
+        ext_i.intersection(com_i).difference(mod_i).rename(ext_i.name)
+    )
+    ext_i = inter_i
```

Comment:
these redefinitions are confusing, we should define everything at the beginning as it was done previously.
It is likely that we need a get_{something} to improve the clarity